specification/storagecache/resource-manager/Microsoft.StorageCache/preview/2020-08-13-aml-preview/storagecache.json

```diff
@@ -0,0 +1,2623 @@
+      "type" : "string",
+      "pattern": "^[0-9a-zA-Z][-0-9a-zA-Z_]{0,78}[0-9a-zA-Z]$",
+      "minLength": 2,
+      "maxLength": 80
+    },
+    "amlFilesystemsListResult": {
+      "description": "Result of the request to list Caches. It contains a list of Caches and a URL link to get the next set of results.",
+      "properties": {
+        "nextLink": {
+          "description": "URL to get the next set of Cache list results, if there are any.",
+          "type": "string"
+        },
+        "value": {
+          "description": "List of Caches.",
+          "items": {
+            "$ref": "#/definitions/amlFilesystem"
+          },
+          "type": "array"
+        }
+      },
+      "type": "object"
+    },
+    "amlFilesystemHealth": {
+      "description": "An indication of amlFilesystem health. Gives more information about health than just that related to provisioning.",
+      "properties": {
+        "state": {
+          "description": "List of amlFilesystem health states.",
+          "enum": [
+            "Unknown",
+            "Unavailable",
+            "Available",
+            "Degraded",
+            "Transitioning",
+            "Maintenance"
+          ],
+          "x-ms-enum": {
+            "name": "amlFilesystemHealthStateType",
+            "modelAsString": true
+          },
+          "type": "string"
+        },
+        "statusDescription": {
+          "description": "Describes of state.",
+          "type": "string"
+        }
+      }
+    },
+    "amlFilesystemEncryptionSettings": {
+      "description": "amlFilesystem encryption settings.",
+      "properties": {
+        "keyEncryptionKey": {
+          "x-ms-secret": true,
+          "$ref": "#/definitions/KeyVaultKeyReference",
+          "description": "Specifies the location of the encryption key in Key Vault."
+        }
+      }
+    },
+    "amlFilesystemHSMSettings": {
+      "description": "amlFilesystem HSM settings.",
+      "properties": {
+        "container": {
+          "description": "Container used for the amlFilesystem.",
+          "$ref": "#/definitions/URLString"
+        },
+        "importPrefix" : {
+          "description": "Path/Prefix for blobs in above container.  Only blobs containing this prefix get pre-hydrated into the cluster namespace",
+          "type" : "string"
+        },
+        "exportPrefix" : {
+          "description": "Path/Prefix in above blob container where data gets exported when an archive task runs",
+          "type" : "string"
+        }
+      }
+    },
+    "amlFilesystemNamespaceHydrationStatus" : {
+      "description" : "Information about the amlFilesystem namespace hydration status",
+      "type" : "object",
+      "readOnly" : true,
+      "properties": {
+        "state" : {
+          "description" : "The state of the namespace hydration",
+          "type" : "string",
+          "enum" : [
+            "NotStarted",
```

Comment:
I'd like something distinct from NotStarted for the case where the user has not configured HSM - how about NotConfigured?

Reply:
Adding NotConfigured.

Will NotStarted be needed?

Reply:
I believe we need both. NotConfigured means the user didn't ask for hydration. NotStarted means the user asked for it but it has not yet started.

Reply:
ok.  They're both there in the current commit.